code/modules/client/preferences.dm

```diff
@@ -294,8 +296,6 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	character_preview_view.update_body()
 	character_preview_view.display_to(user)
 
-	return character_preview_view
```

Comment:
Instead of doing this, make the `character_preview_var` here a fresh variable and continue to return it.

Reply:
I think there's a real bug we're hiding here of not deleting the old character_preview_view btw

Reply:
Right. I'll try to replicate the runtime in pure TG and write back.

Could a problem be accessing Character preference UI before the server loads? It seems to be the reason on skyrat. Anyway, will report back with more data, thanks